ios/brave-ios/Sources/Brave/Frontend/UserContent/UserScripts/Scripts_Dynamic/ScriptHandlers/Paged/BraveSearchResultAdScriptHandler.swift

```diff
@@ -0,0 +1,123 @@
+// Copyright 2024 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import BraveCore
+import BraveShared
+import os.log
+import WebKit
+
+class BraveSearchResultAdScriptHandler: TabContentScript {
+  private struct SearchResultAdResponse: Decodable {
+    struct SearchResultAd: Decodable {
+      let creativeInstanceId: String
+      let placementId: String
+      let creativeSetId: String
+      let campaignId: String
+      let advertiserId: String
+      let landingPage: URL
+      let headlineText: String
+      let description: String
+      let rewardsValue: String
+      let conversionUrlPatternValue: String?
+      let conversionAdvertiserPublicKeyValue: String?
+      let conversionObservationWindowValue: Int?
+    }
+
+    let creatives: [SearchResultAd]
+  }
+
+  fileprivate weak var tab: Tab?
+
+  init(tab: Tab) {
+    self.tab = tab
+  }
+
+  static let scriptName = "BraveSearchResultAdScript"
+  static let scriptId = UUID().uuidString
+  static let messageHandlerName = "\(scriptName)_\(messageUUID)"
+  static let scriptSandbox: WKContentWorld = .page
```

Comment:
Will the script work if we inject it into `.defaultClient` content world? `.page` should be fine since we only inject it on brave search domains, just curious if we can avoid injection into the page context.

Reply:
yes, `.defaultClient` content world works fine here as we still able to access. Thanks for finding this. I will change this to:
`static let scriptSandbox: WKContentWorld = .page `

Reply:
Done